### PR TITLE
[13.0][BKP][ADD]backport `endpoint` modules

### PR DESCRIPTION
**Pull request by GuillemCForgeFlow**
Depends on https://github.com/OCA/server-tools/pull/3098

- Backport [endpoint_route_handler](https://github.com/OCA/web-api/tree/14.0/endpoint_route_handler) from v14.0
- Backport [endpoint](https://github.com/OCA/web-api/tree/14.0/endpoint) from v14.0
- Backport [edi_endpoint_oca](https://github.com/OCA/edi/blob/14.0/edi_endpoint_oca) from v14.0 and rename it to `edi_endpoint` to align with `edi_webservice` module

cc @ForgeFlow

### CODE REVIEW

**Comment by GuillemCForgeFlow** [comment]
Hi @simahawk, I'm trying to backport the `endpoint` modules here in the `edi` repo for v13. Would it be possible for you to review https://github.com/OCA/server-tools/pull/3098 and perhaps merge, so I can rebase here and check if everything is working fine.

Thank you in advance 🙂

File: endpoint_route_handler/models/endpoint_route_sync_mixin.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            # and avoid multiple env restarts in case of multiple edits
+            # on one or more records in a row.
+            self._add_after_commit_hook(self.ids)
+        return res
+
+    @api.model
+    def _add_after_commit_hook(self, record_ids):
+        self.env.cr.postcommit.add(
+            partial(self._handle_registry_sync_post_commit, record_ids),
+        )
```
Comment (JordiMForgeFlow) [inline endpoint_route_handler/models/endpoint_route_sync_mixin.py]:
```suggestion
        self.env.cr.after(
            "commit",
            partial(self._handle_registry_sync_post_commit, record_ids),
        )
```

same change should be applied wherever postcommit is used

**Comment by simahawk** [comment]
@GuillemCForgeFlow base modules should go to web-api. We can create a 13.0 branch there.
As for the edi_endpoint module it could stay here.

**Comment by simahawk** [comment]
There you go https://github.com/OCA/repo-maintainer-conf/pull/53

**Comment by simahawk** [comment]
Et voilá https://github.com/OCA/web-api/tree/13.0
Please fix the little pre-commit issue in a separate PR :pray:

**Comment by GuillemCForgeFlow** [comment]
@simahawk will do, thank you for that 🙂

**Comment by GuillemCForgeFlow** [comment]
closing this one, as the respective PRs have been created:
https://github.com/OCA/web-api/pull/67
https://github.com/OCA/edi/pull/1062

**Pull request closed without merging**